node/network/bridge/src/lib.rs

```diff
@@ -1536,4 +1544,32 @@ mod tests {
 			}
 		});
 	}
+
+
+	#[test]
+	fn verify_coverage() {
+		for msg in spread_event_to_subsystems(NetworkBridgeEvent::PeerDisconnected(PeerId::random())) {
+			match msg {
+				AllMessages::CandidateValidation(_) => unreachable!("Not interested in network events"),
+				AllMessages::CandidateBacking(_) => unreachable!("Not interested in network events"),
+				AllMessages::CandidateSelection(_) => unreachable!("Not interested in network events"),
+				AllMessages::ChainApi(_) => unreachable!("Not interested in network events"),
+				AllMessages::CollatorProtocol(_) => unreachable!("Not interested in network events"),
+				AllMessages::StatementDistribution(_) => {}
+				AllMessages::AvailabilityDistribution(_) => {}
+				AllMessages::AvailabilityRecovery(_) => unreachable!("Not interested in network events"),
+				AllMessages::BitfieldDistribution(_) => {}
+				AllMessages::BitfieldSigning(_) => unreachable!("Not interested in network events"),
+				AllMessages::Provisioner(_) => unreachable!("Not interested in network events"),
+				AllMessages::PoVDistribution(_) => {}
+				AllMessages::RuntimeApi(_) => unreachable!("Not interested in network events"),
+				AllMessages::AvailabilityStore(_) => unreachable!("Not interested in network events"),
+				AllMessages::NetworkBridge(_) => unreachable!("Not interested in network events"),
+				AllMessages::CollationGeneration(_) => unreachable!("Not interested in network events"),
+				AllMessages::ApprovalVoting(_) => unreachable!("Not interested in network events"),
+				AllMessages::ApprovalDistribution(_) => {}
+				// add new variants as needed
+			}
+		}
+	}
```

Comment:
how is it going to make sure we didn't forget about a message? my understanding is that it will only panic if we accidentally forwarded a wrong message type, but not if we forgot about the right one?

Reply:
That part requires a counter, but I wanted to first see if that can be remedied by a procmacro.

If one forgets to add a message here, it won't even compile, and the unit test calls exactly one `fn`.

Reply:
well, you can add it here, but not in `spread_event_to_subsystems` (`multiply_message`) and it will compile :P

Reply:
yeah, counter sounds workable, like https://github.com/paritytech/polkadot/blob/0658b3d2dce54d497d394419f046df198ce2b939/node/overseer/src/lib.rs#L2772